kernel-scriptlets/cert-script

```diff
@@ -60,16 +60,21 @@ case $op in
 	if [ -n "$ca_check" ] && mokutil -h | grep -q "ignore-keyring"; then
 	    MOK_ARGS="${MOK_ARGS} --ignore-keyring"
 	fi
-	# XXX: Only call mokutil if UEFI and shim are used
-	for cert in $certs; do
-	    cert="/etc/uefi/certs/${cert}.crt"
-	    run_mokutil --import "$cert" --root-pw ${MOK_ARGS}
-	    rc=$?
-	    if [ $rc != 0 ] ; then
-		script_rc=$rc
-		echo "Failed to import $cert" >&2
-	    fi
-	done
+	# XXX: Only call mokutil if UEFI and shim are used and efivarfs is writeable
+	test -w /sys/firmware/efi/efivars
+	if [ $? == 0 ] ; then
+	    for cert in $certs; do
+		cert="/etc/uefi/certs/${cert}.crt"
+		run_mokutil --import "$cert" --root-pw ${MOK_ARGS}
+		rc=$?
+		if [ $rc != 0 ] ; then
+		    script_rc=$rc
+		    echo "Failed to import $cert" >&2
+		fi
+	    done
+	else
+	    test -d /sys/firmware/efi/efivars && echo "WARNING: cannot update MOK variables since /sys/firmware/efi/efivars is not writeable" 1>&2
```

Comment:
You shouldn't use `&&` constructs in these scripts, because the compound statement might have an error exit status. Please write

    [ ! -d /sys/firmware/efi/efivars ] || echo ...

instead (any specific reason why you use `test`?

Reply:
Well, the line just says "if it's a directory, echo, otherwise, don't".

Also, your line is also a compound statement (it just uses `||` instead of `&&`) and is equivalent to `test ! -d /sys/firmware/efi/efivars || echo ...`. So, I don't really get the difference in meaning here.

As for why I use `test`:
- I find it more readable
- it's easier to find documentation about it in case you don't know what it means (and yes, I consider things like that important); seriously, try to search for something like that on a search engine or a manpage

Reply:
We use `[` everywhere else, it's POSIX and well-established shell programming practice. But I'm not religious on this matter. 

I am religious wrt `&&` though. 

> Also, your line is also a compound statement (it just uses || instead of &&) and is equivalent to `test ! -d /sys/firmware/efi/efivars || echo ...`

    test ! -d $DIR || echo blah
    if test -d $DIR; then echo blah; fi
    test -d $DIR && echo blah

These statements are logically equivalent, and would be equivalent in most programming languages. But in shell, there's a difference. The first 2 variants always have non-zero exit status. But the last one has non-zero exit status if `$DIR` doesn't exist. If the shell is run with the `-e` flag, the script would abort immediately. This is why `&&` constructs should be avoided.

Reply:
About `test`: it's also part of POSIX (https://pubs.opengroup.org/onlinepubs/9699919799/nframe.html > "Shell & Utilities" > "Utilities" > "test"). And well, as I said, I like programs (and scripts) to be accessible to beginners and as such things should be easy to find in case you don't know about them (and well, test is easier to find than the other construct).

I will change the one about `&&`, thanks for the explanation.

Reply:
> About test: it's also part of POSIX

Sure. And we use both `test` and `[` in this code, so although I clearly favor `[`, I said I am not religious on this matter.